_posts/2024-05-31-friends-dont-let-friends-package-alone-pyconus-24.md

```diff
@@ -0,0 +1,282 @@
+</figure>
+
+### The evolution of the pyOpenSci Python packaging guide
+Years later, what started out as a generic Python development guide, was
+transformed into what’s now the [pyOpenSci Python package guide](https://www.pyopensci.org/python-package-guide/).
+
+* A guidebook that has 58 contributors as I’m writing this post today.
+* A guidebook that has color and graphics and that leads Pythonistas down a
+  singular, opinionated path to Python packaging success.
+* A guidebook that breaks down pieces and processes into their fundamentals and
+  explains them using digestible words.
+
+The pyOpenSci path is not the only path. It is just a path that works and
+enables early wins for scientists and others using Python.
+
+If you're reading this and you have your own path, that's ok too. But if you
+want to worry less about decisions and have a definitive way of doing things
+that follows modern standards -
+
+we've got you.
+
+### A packaging guidebook created by community
+
+This guidebook was created by a community of people with beginner to expert packaging experience with the goal of building consensus
+around a single approach to pure Python packaging that would **yield user success**.
+
+Together we created a guidebook that acknowledges a suite of fantastic tools across the ecosystem while
+still leading users down a streamlined path.
+
+<figure>
+  <img src="{{ site.baseurl }}/images/blog/2024/june/pyopensci-packaging-guide-overview.png" alt="Image with the pyOpenSci flower logo in the upper right hand corner. The image shows the packaging lifecycle. The graphic shows a high level overview of the elements of a Python package. the inside circle has 5 items - user documentation, code/api, test suite, contributor documentation, project metadata / license / readme. In the middle of the circle is says maintainers and has a small icon with people. on the outside circle there is an arrow and it says infrastructure." style="width: 70%;" />
+  <figcaption><i class="fa-solid fa-heart-pulse"></i> The generalized scientific Python package development process. <i class="fa-solid fa-heart-pulse"></i>.</figcaption>
+</figure>
+
+
+## Getting accepted to talk about packaging...  <i class="fa-solid fa-ghost"></i>
+
+I was thrilled and terrified to be accepted to give a talk on the process of creating the pyOpenSci Python packaging guidebook at pyCon US in Pittsburgh, Pennsylvania. My talk was called:
+
+“Friends don’t let friends package alone”
+
+It was all about the process of navigating the Python packaging ecosystem and
+then creating the guide. While you might assume such a talk about be technical,
+in reality, it was all about people.
+
+### Me and the child-like imposter inside
+
+Giving this talk was a big deal for me. Imposter syndrome—the feeling that I
```

Comment:
```suggestion
Giving this talk was a big deal for me. Imposter syndrome—the feeling that I
```

Reply:
one thing to mention is that the use of hyphens (essentially in place of em-dashes) is inconsistent. I'm not sure that the --- gets converted to an em-dash in Markdown, but for consistency we should be doing either words connected by the dash (syndrome-the) or with spaces (syndrome - the).

with an em-dash, the correct way is to have the dash "touch" the words, without a space. but if we don't get a rendered em-dash (the longest hyphen), then spaces make more sense, as they increase readability.

Reply:
@kierisi i always get confused with em dashes - i know there are different types of dashes. i don't know when ti use which one. is em dash the longer one so it should always touch the words? if that is the case i can fix throughout locally.

Reply:
I only know because we ended up doing a deep dive on this when we wrote Data Science in Education Using R! so en dashes are *generally* used for ranges in numbers and dates, and em dashes are a replacement for commas or parentheses. the way I remember it is that an eN dash is for Numbers. in both cases (along with hyphens!) the dash should be touching both words.

Reply:
ok - that's helpful. It looks like to use an em-dash i'd have to add the unicode value

&#151; 

i can try that? otherwise markdown is not fancy and doesn't know about typography things. Let me see how unicode renders!

Reply:
sounds good! honestly I think an en-dash is fine, it's more that the style is consistent!

Reply:
actually just pasting the character in also works - `—` that's much easier.